root_pandas/readwrite.py

```diff
@@ -147,6 +148,18 @@ def do_flatten(arr):
         arr = append_fields(arr_, '__array_index', idx, usemask=False, asrecarray=True)
         return arr
 
+    def remove_high_dimensions(arr):
+        allowed_dimensions = [0]
+        first_row = arr[0]
+        good_cols = [True if x.ndim in allowed_dimensions else False for x in first_row]
+        col_names = np.array(list(arr.dtype.names))
+        good_names = col_names[np.array(good_cols)]
+        bad_names = col_names[np.array([not x for x in good_cols])]
+        for bad_name in bad_names:
+            warnings.warn("Dropped {bad_name} branch because dimension is unfit for DataFrame"
```

Comment:
Would be better to only emit a single warning in my opinion. How about `"Ignored the following non-scalar branches: {bad_names}".format(bad_names=", ".join(bad_names))`?